skyrim-platform/src/platform_se/skyrim_platform/EventsApi.cpp

```diff
@@ -122,7 +122,13 @@ class Hook
   }
 
   // Chakra thread only
-  void AddHandler(const Handler& handler) { handlers.push_back(handler); }
+  uint32_t AddHandler(const Handler& handler)
+  {
+    handlers.emplace(hCounter, handler);
+    return hCounter++;
+  }
+
+  void RemoveHandler(const uint32_t& id) { handlers.erase(id); }
```

Comment:
Users might call `remove` during the `enter`/`leave` execution. It will break `for (auto& hp : handlers)` loop and crash the game. Please fix that: `throw std::runtime_error(...)` if we are trying to remove a hook handler in `enter`/`leave` calls.

Btw it seems that in the original version we were able to crash in a similar way: by using `add` inside `enter`/`leave`. For consistency please fix that too.

Reply:
using `add/remove` inside `enter/leave` will now throw an exception. It's still not actually completely thread safe though, `enter/leave` while `add/remove` is running is going to cause problems.

Reply:
I didn't see a problem here: we cannot enter/leave while adding a hook. Approving this PR. If I'm wrong with it and there is a real problem, feel free to leave an issue, etc
@nn-tran